src/VisualStudio/Core/Def/Implementation/ProjectSystem/RunningDocumentTableEventTracker.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    /// <summary>
+    /// Class to register with the RDT and forward RDT events.
+    /// Handles common conditions and delegates implementation to the <see cref="IRunningDocumentTableEventListener"/>
+    /// </summary>
+    internal sealed class RunningDocumentTableEventTracker : IVsRunningDocTableEvents3, IDisposable
+    {
+        private bool _isDisposed = false; // To detect redundant calls
+
+        private readonly ForegroundThreadAffinitizedObject _foregroundAffinitization;
+        private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
+        private readonly IVsRunningDocumentTable4 _runningDocumentTable;
+        private readonly IRunningDocumentTableEventListener _runningDocumentTableEventListener;
```

Comment:
>_runningDocumentTableEventListener [](start = 60, length = 34)

I think we can just call the field `_listener`. The rest is obvious from context.